ext/standard/head.c

```diff
@@ -248,24 +224,61 @@ PHP_FUNCTION(setcookie)
 	if (expires_or_options) {
 		if (Z_TYPE_P(expires_or_options) == IS_ARRAY) {
 			if (UNEXPECTED(ZEND_NUM_ARGS() > 3)) {
-				php_error_docref(NULL, E_WARNING, "Cannot pass arguments after the options array");
-				RETURN_FALSE;
+				zend_argument_count_error("%s(): Expects exactly 3 arguments when argument #3 "
+					"($expires_or_options) is an array", get_active_function_name());
+				RETURN_THROWS();
 			}
 			php_head_parse_cookie_options_array(expires_or_options, &expires, &path, &domain, &secure, &httponly, &samesite);
+			if (path && strpbrk(ZSTR_VAL(path), ",; \t\r\n\013\014") != NULL) { /* man isspace for \013 and \014 */
+				zend_value_error("%s(): Argument #3 ($expires_or_options[\"path\"]) cannot contain "
+					ILLEGAL_COOKIE_CHARACTER, get_active_function_name());
+				goto cleanup;
+				RETURN_THROWS();
```

Comment:
This RETURN_THROWS() doesn't make much sense, as it cannot be executed after the `goto cleanup`.